st2common/st2common/models/api/base.py

```diff
@@ -71,7 +70,22 @@ def __json__(self):
         return vars(self)
 
     def validate(self):
-        VALIDATOR(getattr(self, 'schema', {})).validate(vars(self))
+        """
+        Perform validation and return cleaned object on success.
+        """
+        schema = getattr(self, 'schema', {})
+        attributes = vars(self)
+
+        cleaned = util_schema.validate(instance=attributes, schema=schema,
+                                       cls=util_schema.CustomValidator, use_default=True)
+
+        # Note: For backward compatibility reasons, we re-assign attributes one the current
```

Comment:
I made the "return new cleaned instance on validate" change in 37f3f8e7e1cbf26983a53ea5ce2aead53e8d86d7.

All the tests pass and I grepped for "validate" in the code base and I didn't find any more usages of it. In any case, we should still keep an eye on it in case I missed something and a regression pops up in the future.